pkg/alerts/prometheus.go

```diff
@@ -7,22 +7,39 @@ import (
 	"sort"
 	"time"
 
-	"github.com/prometheus/client_golang/api"
+	papi "github.com/prometheus/client_golang/api"
 	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/prometheus/common/model"
 )
 
-// PrometheusActiveAlerts returns a list of names of active (e.g. pending or firing) alerts, filtered
-// by the supplied regexp.
-func PrometheusActiveAlerts(prometheusURL string, filter *regexp.Regexp) ([]string, error) {
-	client, err := api.NewClient(api.Config{Address: prometheusURL})
+// PromClient is a wrapper around the Prometheus Client interface and implements the api
+// This way, the PromClient can be instantiated with the configuration the Client needs, and
+// the ability to use the methods the api has, like Query and so on.
+type PromClient struct {
+	papi papi.Client
+	api  v1.API
+}
+
+// New creates a new client to the Prometheus API.
+// It returns an error on any problem.
+func New(conf papi.Config) (*PromClient, error) {
```

Comment:
this is looking good, thanks!

I think reading through this, it would be clearer if we rename `New` to `NewPromClient`. Sorry for all the change requests, I think we're almost there.

Reply:
I had the same feeling. I proceeded with it, because of the name of instantiation promClient. If it were a prometheus package, then it would read good enough, like prometheus.New(). 
I'll change it this evening. Thanks for the feedback and help.